src/Plugin/Field/FieldFormatter/StrawberryBaseFormatter.php

```diff
@@ -0,0 +1,246 @@
+    $this->iiif_pub_server = $config->get('pub_server_url');
+    $this->iiif_int_server = $config->get('int_server_url');
+  }
+
+  //todo @marlo -- use config factory, Docs say best practice is to use dep injection, not \Drupal::
+  public static function defaultSettings() {
+    return [
+      'iiif_base_url' => \Drupal::config('format_strawberryfield.iiif_settings')->getOriginal('pub_server_url', FALSE),
+      'iiif_base_url_internal' => \Drupal::config('format_strawberryfield.iiif_settings')->getOriginal('int_server_url', FALSE),
+      ] + parent::defaultSettings();
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function IiifSettingsForm(array $form, FormStateInterface $form_state) {
+    return [
+      'iiif_base_url' => [
+        '#type' => 'url',
+        '#title' => $this->t('Base Public accesible URL of your IIIF Media Server'),
+        '#default_value' => $this->getSetting('iiif_base_url'),
+        '#required' => TRUE
+      ],
+      'iiif_base_url_internal' => [
+        '#type' => 'url',
+        '#title' => $this->t('Base URL of your IIIF Media Server accesible from inside this Webserver'),
+        '#default_value' => $this->getSetting('iiif_base_url_internal'),
+        '#required' => TRUE,
+        '#element_validate' => [[$this, 'validateIiifUrl']],
+      ],
+      'iiif_reset_button' => [
+        '#type' => 'button',
+        '#name' => 'iiif_reset_button',
+        '#value' => t('Reset to IIIF Defaults'),
+        '#executes_submit_callback' => TRUE,
+        '#submit' => [[$this, 'resetIiifDefaults']],
+      ],
+    ];
+  }
+
+
+  public function validateIiifUrl(array $form, FormStateInterface $form_state) {
+    // todo: @marlo -- also broken, same issue with non-updated form_state
+    error_log(var_export( $form_state->get('iiif_base_url_internal'), true));
+    if ($form_state->get('iiif_base_url_internal')) {
+    try {
+      $response = $this->httpClient
+        ->head(
+          $form_state->get('iiif_base_url_internal')
+        );
+    }
+    catch(ConnectException $exception) {
+      $responseMessage = $exception->getMessage();
+      $form_state->setErrorByName('iiif_base_url_internal', t("We could not contact your Public IIIF server: @error", [
+        '@error' => $responseMessage
+      ]));
+    }
+    catch(ClientException $exception) {
+      $responseMessage = $exception->getMessage();
+      $form_state->setErrorByName('iiif_base_url_internal', t("We could not contact your Public IIIF server: @error", [
+        '@error' => $responseMessage
+      ]));
+    }
+    catch (ServerException $exception) {
+      $responseMessage = $exception->getMessage();
+      $form_state->setErrorByName('iiif_base_url_internal', t("We could not contact your Public IIIF server: @error", [
+        '@error' => $responseMessage
+      ]));
+    }
+    }
+  }
+
+
+  /**
+   * {@inheritdoc}
+   */
+  public function resetIiifDefaults(array $form, FormStateInterface $form_state) {
+    $this->setSettings([
```

Comment:
I feel(nor sure, just out of my head) setSettings will try to set all Settings at once. So if your Concrete class has its own settings, then this will write all settings and remove the other ones. Try setSetting, feels more precise.